src/map.cpp

```diff
@@ -863,9 +863,9 @@ int_fast32_t AStarNodes::getTileWalkCost(const Creature& creature, const Tile* t
 // Floor
 Floor::~Floor()
 {
-	for (uint32_t i = 0; i < FLOOR_SIZE; ++i) {
-		for (uint32_t j = 0; j < FLOOR_SIZE; ++j) {
-			delete tiles[i][j];
+	for (auto& tile : tiles) {
+		for (auto& j : tile) {
```

Comment:
That won't work since ptr is the pointer to the tile in the first index, the tile stored at the second index is not necessarily allocated at the the memory address of the tile allocated at the first index + sizeof(Tile), in fact it might be NULL but your code will try to delete at some other address anyway, also if the tile at the first index is NULL then it'll most likely segfault at the second index anyway because then you'll be deleting at 0+sizeof(Tile). This destructor is rarely called, if it's called at all, that it's not worth trying to do something fancy although I understand what you're trying to do.